src/horizontal-pod-autoscaler.ts

```diff
@@ -0,0 +1,865 @@
+    return new Metric({
+      type: MetricSourceType.CONTAINER_RESOURCE,
+      containerResource: {
+        name: ResourceType.CPU,
+        container: options.container.name,
+        target: options.target._toKube(),
+      },
+    });
+  }
+
+  /**
+   * Metric that tracks the Memory of a container. This metric
+   * will be tracked across all pods of the current scale target. Only a single
+   * container may be tracked in each pod. In other words, a pod that hosts two
+   * of the same container, will be ignored.
+   *
+   */
+  public static containerMemory(options: MetricContainerResourceOptions): Metric {
+    return new Metric({
+      type: MetricSourceType.CONTAINER_RESOURCE,
+      containerResource: {
+        name: ResourceType.MEMORY,
+        container: options.container.name,
+        target: options.target._toKube(),
+      },
+    });
+  }
+
+  /**
+   * Metric that tracks the volume size of a container. This metric
+   * will be tracked across all pods of the current scale target. Only a single
+   * container may be tracked in each pod. In other words, a pod that hosts two
+   * of the same container, will be ignored.
+   *
+   */
+  public static containerStorage(options: MetricContainerResourceOptions): Metric {
+    return new Metric({
+      type: MetricSourceType.CONTAINER_RESOURCE,
+      containerResource: {
+        name: ResourceType.STORAGE,
+        container: options.container.name,
+        target: options.target._toKube(),
+      },
+    });
+  }
+
+  /**
+   * Metric that tracks the local ephemeral storage of a container. This metric
+   * will be tracked across all pods of the current scale target. Only a single
+   * container may be tracked in each pod. In other words, a pod that hosts two
+   * of the same container, will be ignored.
+   *
+   */
+  public static containerEphemeralStorage(options: MetricContainerResourceOptions): Metric {
+    return new Metric({
+      type: MetricSourceType.CONTAINER_RESOURCE,
+      containerResource: {
+        name: ResourceType.EPHEMERAL_STORAGE,
+        container: options.container.name,
+        target: options.target._toKube(),
+      },
+    });
+  }
+
+  /**
+   * A global metric that is not associated with any Kubernetes object.
+   * Allows for autoscaling based on information coming from components running outside of
+   * the cluster.
+   *
+   * Use case:
+   * * Scale up when the length of an SQS queue is greater than 1000 messages.
+   * * Scale down when an outside load balancer's queries are less than 10 per second.
+   */
+  public static external(options: MetricOptions): Metric {
+    return new Metric({
+      type: MetricSourceType.EXTERNAL,
+      external: {
+        metric: {
+          name: options.name,
+          selector: options?.labelSelector?._toKube(),
+        },
+        target: options.target._toKube(),
+      },
+    });
+  }
+
+  /**
+  * Metric that describes a kubernetes object
```

Comment:
Not entirely accurate, at least not based on the API.

```suggestion
  * Metric that describes a metric of a kubernetes object
```